src/tests/Microsoft.PowerFx.Interpreter.Tests/RecalcEngineTests.cs

```diff
@@ -593,6 +593,49 @@ public void FunctionPrecedenceTest(string script, double expected)
             Assert.Equal(expected, result.AsDouble());
         }
 
+        [Theory]
+
+        // Return value with side effectful UDF
+        [InlineData(
+            "F1(x:Number) : Number = { Set(a, x); a+1; };",
+            "F1(123)",
+            false,
+            124,
+            true)]
+
+        // Mismatch return value with side effectful UDF
+        [InlineData(
+            "F1(x:Number) : Boolean = { Set(a, x); Today(); };",
+            null,
+            true,
+            0,
+            true)]
+
+        public void ImperativeUserDefinedFunctionTest(string udfExpression, string expression, bool expectedError, double expected = 0, bool allowSideEffects = false)

```

Comment:
> = 0 [](http://example.com/codeflow?start=130&length=4)

No need to define the default values for these arguments as they are passed in all cases